android/src/org/mozilla/firefox/vpn/InAppPurchase.kt

```diff
@@ -306,10 +317,12 @@ class InAppPurchase private constructor(ctx: Context) :
     fun processPurchases(purchases: MutableList<Purchase>?) {
         if (purchases == null) {
             Log.d(TAG, "onPurchasesUpdated: null purchase list")
+            VPNUtils.sendGleanEvent(GleanEvent.iapPurchaseUpdateFailed)
```

Comment:
I do not believe this is a failure state. I think the processPurchases call back can get fired sometimes even when there's nothing to do and it's just a "log and move on" type situation rather than an error (which is why I did a `Log.d`) but let's check the docs.

Reply:
I'm okay with keeping it though as it is useful data as to whether this happens a lot. If we do keep it; I'd like it to be a separate event and not share a name so we can distinguish the error states.